test/e2e/util.go

```diff
@@ -249,16 +246,28 @@ func lookupCNAME(host, server string) (string, error) {
 	return cname.Target, nil
 }
 
-func equalFQDN(name1, name2 string) bool {
-	index1 := strings.LastIndex(name1, ".")
-	index2 := strings.LastIndex(name2, ".")
-	if index1 != len(name1)-1 {
-		name1 += "."
+func lookupARecord(host, server string) ([]string, error) {
+	dnsClient := &dns.Client{}
+	message := &dns.Msg{}
+	message.SetQuestion(dns.Fqdn(host), dns.TypeA)
+	response, _, err := dnsClient.Exchange(message, fmt.Sprintf("%s:53", server))
+	if err != nil {
+		return nil, err
 	}
-	if index2 != len(name2)-1 {
-		name2 += "."
+	if len(response.Answer) == 0 {
+		return nil, fmt.Errorf("no IPs found for host %s on %s", host, server)
 	}
-	return name1 == name2
+	var ips []string
+	for _, ans := range response.Answer {
+		if aRec, ok := ans.(*dns.A); ok {
+			ips = append(ips, aRec.A.String())
+		}
```

Comment:
Can we match the behavior of `lookupCNAME` function here? For the case when the record is not `dns.A`.

Reply:
Looks like there are some records in the response which are not A records and that causes the whole lookup to fail. Instead I have added a check if number of IPs is 0.